test/unit/pdf/finance/invoice_report_data_test.rb

```diff
@@ -86,4 +87,40 @@ def setup
     @provider.update_attribute(:org_name, '<ScRipT>alert("address1")</ScRipT>')
     assert_equal @data.provider[0][1], '&lt;ScRipT&gt;alert(&quot;address1&quot;)&lt;/ScRipT&gt;'
   end
+
+  test '#with_logo yields to a block with open file and close is after' do
+    @provider.profile.update(logo: Rack::Test::UploadedFile.new(file_fixture('wide.jpg'), 'image/jpeg', true))
+
+    logo_file = nil
+    @data.with_logo do |logo|
+      logo_file = logo
+      assert logo.is_a? File
+      assert logo.binmode?
+      assert logo.respond_to?(:read)
+    end
+    assert logo_file.closed?
+  end
+
+  test '#with_logo yields nil if logo not set' do
+    @provider.profile.update(logo: nil)
+
+    @data.with_logo do |logo|
+      assert_nil logo
+    end
+  end
+
+  test '#with_logo calls S3 URL if S3 is used for storage' do
+    default_options = Paperclip::Attachment.default_options
+    Paperclip::Attachment.stubs(default_options: default_options.merge(storage: :s3))
+    @provider.profile.update(logo: Rack::Test::UploadedFile.new(file_fixture('wide.jpg'), 'image/jpeg', true))
+    URI.expects(:open).with(regexp_matches(/\Ahttps.*\/profiles\/logos\/invoice\/wide.png\z/)).returns(File.open(file_fixture('wide.jpg')))
```

Comment:
**IMPORTANT:** this will break once https://github.com/3scale/porta/pull/3065 is merged (`wide.jpg` will be instead of `wide.png`).
Will need to update the tests, depending on which branch is merged earlier.